build/tasks/build.js

```diff
@@ -81,7 +84,7 @@ module.exports = function( grunt ) {
 
 			// Remove define wrappers, closure ends, and empty declarations
 			contents = contents
-				.replace( /define\([^{]*?{/, "" )
+				.replace( /define\([^{]*?{\s*(?:"use strict";)?/, "" )
```

Comment:
OK, then `(?:"use strict";)` -> `(?:("|')use strict\1;` should work. I think there are enough people on the jQuery Core team with extremely strong preference to not omit semi-colons that I don't expect it to ever happen but if you feel strongly about being secure here `(?:("|')use strict\1(?:;|)` would work as well.